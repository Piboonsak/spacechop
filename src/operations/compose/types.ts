import * as t from 'runtypes';

const ComposeConfig = t.Record({
  source: t.String,
}).And(t.Partial({
  params: t.Dictionary(t.String, 'string'),
}));

export type ComposeConfig = t.Static<typeof ComposeConfig>;
export default ComposeConfig;
