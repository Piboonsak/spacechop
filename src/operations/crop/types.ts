import * as t from 'runtypes';
import Gravity from './../Gravity';

const CropConfig = t.Partial({
  width: t.Number,
  height: t.Number,
  gravity: Gravity,
}).withConstraint((n) => n && !!(n.width || n.height));

export type CropConfig = t.Static<typeof CropConfig>;
export default CropConfig;
