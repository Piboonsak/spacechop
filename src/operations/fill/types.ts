import * as t from 'runtypes';
import Gravity from './../Gravity';
import PostiveNumber from './../PositiveNumber';

const FillConfig = t.Record({
  width: PostiveNumber,
  height: PostiveNumber,
}).And(t.Partial({
  gravity: Gravity,
}));

export type FillConfig = t.Static<typeof FillConfig>;
export default FillConfig;
