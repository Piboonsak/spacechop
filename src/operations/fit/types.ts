import * as t from 'runtypes';
import Gravity from './../Gravity';

const FitConfig = t.Partial({
  width: t.Number,
  height: t.Number,
  gravity: Gravity,
});


export type FitConfig = t.Static<typeof FitConfig>;
export default FitConfig;
