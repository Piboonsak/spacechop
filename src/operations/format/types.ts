import * as t from 'runtypes';
import Format from './../../types/Format';

const FormatConfig = t.Record({
  type: Format,
});

export type FormatConfig = t.Static<typeof FormatConfig>;
export default FormatConfig;
