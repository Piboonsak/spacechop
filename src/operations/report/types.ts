import * as t from 'runtypes';
import ReportFormat from './../../types/ReportFormat';

const ReportConfig = t.Partial({
  format: ReportFormat,
  compact: t.Boolean,
});

export type ReportConfig = t.Static<typeof ReportConfig>;
export default ReportConfig;
