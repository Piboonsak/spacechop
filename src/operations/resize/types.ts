import * as t from 'runtypes';
import PositiveNumber from '../PositiveNumber';

const ResizeConfig = t.Record({
  width: PositiveNumber,
  height: PositiveNumber,
});

export type ResizeConfig = t.Static<typeof ResizeConfig>;
export default ResizeConfig;
