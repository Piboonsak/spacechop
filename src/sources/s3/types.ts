import * as t from 'runtypes';

const RuntypeS3Config = t.Record({
  s3: t.Record({
    access_key_id: t.String.withConstraint((n) => n && n.length > 0),
    secret_access_key: t.String.withConstraint((n) => n && n.length > 0),
    region: t.String.withConstraint((n) => n && n.length > 0),
    bucket_name: t.String.withConstraint((n) => n && n.length > 0),
    path: t.String,
  }),
});

export interface S3Config {
  access_key_id: string;
  secret_access_key: string;
  region: string;
  bucket_name: string;
  path: string;
}

export default RuntypeS3Config;



