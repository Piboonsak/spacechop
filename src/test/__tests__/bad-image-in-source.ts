import path from 'path';
import extractPathParams from '../../lib/extractPathParams';
import { Config } from '../../types/Config';
import { requestHandler } from './../../spacechop';
import assetsFolder from './../assets/dirname';
import { Request, Response } from './../utils/expressMocks';

/**
 * Tests in this file use the full spacechop implementation by mocking Express
 * Response and Requests.
 */
describe('Bad image in source', () => {
  const p = '/:preset/:image';
  const config: Config = {
    sources: [{
      volume: {
        root: path.join(assetsFolder, ':image'),
      },
    }],
    paths: [p],
    presets: {
      t_original: {
        steps: [],
      },
    },
  };

  const sources = [
    {
      key: jest.fn(),
      exists: jest.fn(() => Promise.resolve(true)),
      stream: jest.fn(() => Promise.reject()),
    },
  ];

  const params = extractPathParams(p);
  const handler = requestHandler(config, params, sources);

  let request;
  let response;
  beforeAll(async () => {
    request = new Request();
    response = new Response();
    request.setParams(0, 't_original');
    request.setParams(1, 'zero-byte-file');
    await handler(request, response);
  });
  it('should set status 400', () => {
    expect(response.status).toBeCalledWith(400);
    expect(response.end.mock.calls[0]).toMatchSnapshot();
  });
});
