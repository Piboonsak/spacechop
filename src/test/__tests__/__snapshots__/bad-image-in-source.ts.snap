// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bad image in source should set status 400 1`] = `
Array [
  "Bad request",
]
`;

exports[`Image missing in source should set status 404 1`] = `
Array [
  "Could not find image",
]
`;
